src/Ledger/GovernanceActions.lagda

```diff
@@ -23,6 +23,9 @@ open import Tactic.Derive.DecEq
 module Ledger.GovernanceActions (gs : _) (open GovStructure gs) where
 
 2ℚ = 1ℚ Data.Rational.+ 1ℚ
+
+maximum : ℙ ℚ → ℚ
+maximum x = foldl Data.Rational._⊔_ 0ℚ (proj₁ $ finiteness x)
```

Comment:
Maybe add a note that we should use a more generic version of this in the future, i.e.
1. `maximum` defined over any (decidably) ordered type:
```agda
module _ {A : Type ℓ} ⦃ _ : Ord A ⦄ ⦃ _ : DecOrd A ⦄ where

min max : Op₂ A
min x y = if ⌊ x ≤? y ⌋ then x else y
max x y = if ⌊ y ≤? x ⌋ then x else y

minimum maximum : A → List A → A
minimum = foldl min
maximum = foldl max
```
2. `maximumˢ` defined for finite sets by `maximum ∘ proj₁ ∘ finiteness`

Reply:
Yeah, there's an even better way to define `maximumˢ` which uses `Axiom.Set.Sum`. It's the same underlying definition, but it requires commutativity of the operation and gives you a bunch of theorems for that.